packages/astro/src/@types/astro.ts

```diff
@@ -1187,7 +1187,7 @@ interface AstroSharedContext<Props extends Record<string, any> = Record<string,
 	/**
 	 * Redirect to another page (**SSR Only**).
 	 */
-	redirect(path: string, status?: 301 | 302 | 308): Response;
+	redirect(path: string, status?: 301 | 302 | 303 | 307 | 308): Response;
```

Comment:
This makes sense to me. Maybe we could add `300` and `304` here too? Here's the [docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#redirection_messages) for convenience.

We chose only the few status in https://github.com/withastro/astro/pull/4986#discussion_r991544356. Might be good to get thoughts from @matthewp  in case I miss something.

Reply:
300 and 304 don't really make sense with the rest of this signature. The signature allows callers to provide a single URL which fills in the `Location:` header, but:
* [300](https://httpwg.org/specs/rfc9110.html#status.300) says "the server SHOULD generate content in the 300 response containing a list of representation metadata and URI reference(s) from which the user or user agent can choose the one most preferred", which needs more than just a URL, and
* [304](https://httpwg.org/specs/rfc9110.html#status.304) doesn't need a `Location` at all, since it says that the client already has the resource cached.

I'd certainly rather get too many options than too few; just wanted to mention that those don't fit well.

Reply:
Sorry I haven't been catching up on the PR reviews. Thanks for the explanation! I think we can put 300 and 304 aside for now until someone asks for it with a reason. Let's keep it as-is.